identity-wallet/src/state/reducers/dev_mode.rs

```diff
@@ -38,20 +42,248 @@ lazy_static! {
     );
 }
 
-pub async fn set_dev_mode(state: AppState, action: Action) -> Result<AppState, AppError> {
-    if let Some(enabled) = listen::<SetDevMode>(action).map(|payload| payload.enabled) {
-        return Ok(AppState {
-            dev_mode_enabled: enabled,
-            current_user_prompt: None,
-            ..state
-        });
+const PASSWORD: &str = "sup3rSecr3t";
+
+pub async fn load_dev_profile(state: AppState, action: Action) -> Result<AppState, AppError> {
+    info!("Load dev profile: {:?}", action);
+
+    if let Some(dev_profile) = listen::<DevProfile>(action) {
+        match dev_profile.profile {
+            ProfileType::Ferris => return load_ferris_profile().await,
+            ProfileType::Dragon => return load_dragon_profile(state).await,
+            ProfileType::None => {}
+        }
     }
+
     Ok(state)
 }
 
-pub async fn load_dev_profile(_state: AppState, _action: Action) -> Result<AppState, AppError> {
-    info!("load dev profile");
+pub async fn toggle_dev_settings(mut state: AppState, _action: Action) -> Result<AppState, AppError> {
```

Comment:
The function name is now misleading, since it does not "toggle" anything anymore (which means switching between two options). All it does is set the `dev_profile` to `None` in any case. You could as well call it `disable_dev_profile()` or `clear_dev_profile()` now. Also, I think it's quite weird that there can exist a `Some(ProfileType::None)`. How does that reflect in the app exactly? Is there a `dev_profile` now or not?

Reply:
It does toggle, it turn on or off dev settings. It just sets it back to profile none after.

Reply:
Changed.

Reply:
Can you please call it `toggle_dev_mode` to be consistent and not introduce another term: `dev_settings`?